src/functions/scripts.js

```diff
@@ -209,6 +211,28 @@ allDetails.forEach(details => {
   });
 });
 
+function updatePageTitle() {
+  const targetId = window.location.hash.substring(1);
+  if (targetId) {
+    const targetElement = document.getElementById(targetId);
+    if (targetElement) {
+      const headWordElement = targetElement.querySelector('span.head-word');
+      if (headWordElement) {
+        const headWordText = headWordElement.textContent.trim();
+        document.title = `${headWordText} | ${SITENAME}`;
+        return;
+      }
+    }
+  }
+  // else
+  document.title = SITENAME;
+}
+
+document.addEventListener("DOMContentLoaded", updatePageTitle);
+window.addEventListener("hashchange", updatePageTitle);
```

Comment:
These two listeners pull the head-word from the targeted element when the page first loads or when the user manually changes the `url#hash` themselves in the address bar.

Note that this method pulls the head string from the element and so includes the " 1" in e.g. `#loss` while clicking on "loss" in the result does not.  Not sure if you care about this minor discrepancy.